txnkv/txnsnapshot/snapshot.go

```diff
@@ -1025,6 +1005,43 @@ func (s *KVSnapshot) SnapCache() map[string][]byte {
 	return cp
 }
 
+// UpdateSnapshotCache sets the values of cache, for further fast read with same keys.
+func (s *KVSnapshot) UpdateSnapshotCache(keys [][]byte, m map[string][]byte) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	if s.mu.cached == nil {
+		s.mu.cached = make(map[string][]byte, min(len(keys), 8))
+	}
+	for _, key := range keys {
+		val := m[string(key)]
+		s.mu.cachedSize += len(key) + len(val)
```

Comment:
If the cache already contains the key, should it minus the original value length here?